### DISCUSSION TITLE
command line: library loading issue: supposedly 0 genes & 0 proteins in library

### DISCUSSION THREAD

**Discussion by JM-Bader** [vdemichev/DiaNN#345] [Q&A]
I ran a command line search and load a .tsv library that is the result of a previous DIA-NN GUI search. 
the command line feedback tells me first that the spectral library contains  2460 protein isoforms, 1283 protein groups, and 7565 precursors. A few lines later it says that the library contains 0 proteins and 0 genes. The results are empty, 0 genes are identified. 

command line:
![image](https://user-images.githubusercontent.com/98888664/159011868-8d5377f7-848c-4159-b965-c429856245fb.png)

spectral library: 
![image](https://user-images.githubusercontent.com/98888664/159012050-9c25d5f2-d504-4191-822a-2846d48827a7.png)

**Comment by vdemichev** [comment]
Does the FASTA file contain protein names / gene names for the sequence IDs listed in the library, e.g. for Q99453?

**Reply by vdemichev** [reply]
Btw, the results will not be empty, all sequence IDs should be properly recorded.

**Reply by JM-Bader** [reply]
> Does the FASTA file contain protein names / gene names for the sequence IDs listed in the library, e.g. for Q99453?

Yes it does: 

![image](https://user-images.githubusercontent.com/98888664/159127875-15fd66dd-0a66-4a81-afea-25f13c6b60cd.png)

Both the library and fasta file contain human protein entries with UniProt identifiers

**Reply by vdemichev** [reply]
Oh... I see. Just noticed, the FASTA file was just not read correctly. I assume some problem with file access permissions on the Linux system?

**Reply by JM-Bader** [reply]
thanks for the hint. 
THis appears to be similar to the issue with loading the library (even though there was an explicit error message for the library but not for fasta file). 

I now copied the fasta file to the home directory and removed all non-letter characters from the name (to be sure that doesnt cause a problem). 
![image](https://user-images.githubusercontent.com/98888664/159128409-70d2f2cd-c922-4e5a-acb9-fbf90fa35077.png)

When reading both library and fasta from the home directory but raw file from network drive (due to storage space), it appears to work now. 

command: 
![image](https://user-images.githubusercontent.com/98888664/159128560-e72d293d-1dc2-48e9-bb1e-5ee47ca0609d.png)

result:
![image](https://user-images.githubusercontent.com/98888664/159128467-f8f61014-117d-4da1-a098-5a4d0bb29337.png)

Nevertheless, the relationship of the different levels of identifiers of proteoforms is unfortunatelly not clear:
![image](https://user-images.githubusercontent.com/98888664/159128742-66f5cc24-67c4-4ae9-b0b3-f4c82065a24a.png)
It says: Spectral library loaded:  3389 protein isoforms, 1685 protein groups and 11913 precursors in 10533 elution groups.
but also: Library contains 2295 proteins, and 1615 genes
how do the two statements match ? What exactly is meant with 2295 "proteins"?

**Reply by vdemichev** [reply]
In case of Q99453, in DIA-NN terminology 'Q99453' is the sequence ID or isoform, 'PHX2B_HUMAN' is the protein name, 'PHOX2B' is the gene.
